src/main/kotlin/creator/ProjectSetupFinalizerWizardStep.kt

```diff
@@ -110,162 +85,57 @@ interface ProjectSetupFinalizer {
-        val projectJdk = context.projectJdk ?: return true
-        val usingCompatibleJdk = isUsingCompatibleJdk(creator, projectJdk)
-        if (!usingCompatibleJdk) {
-            jdkBox.setInvalidJdk(projectJdk.name)
-        } else {
-            jdkBox.selectedJdk = sdksModel.findSdk(projectJdk.name)
-        }
-        updateUi(usingCompatibleJdk)
-        return usingCompatibleJdk
-    }
-
-    override fun validateChanges(creator: MinecraftProjectCreator, context: WizardContext): Boolean {
-        return isUsingCompatibleJdk(creator, jdkBox.selectedJdk ?: return false)
-    }
-
-    override fun apply(creator: MinecraftProjectCreator, context: WizardContext) {
-        val selectedJdk = jdkBox.selectedJdk
-        if (selectedJdk != null) {
-            context.projectJdk = selectedJdk
-        }
+    interface Factory {
+        fun create(parent: NewProjectWizardStep): ProjectSetupFinalizer
     }
 }
 
-class GradleProjectSetupFinalizer : ProjectSetupFinalizer {
-
-    private val model = SortedComboBoxModel<SemanticVersion>(Comparator.naturalOrder())
-
-    private val propertyGraph = PropertyGraph("GradleProjectSetupFinalizer graph")
-    var gradleVersionProperty = propertyGraph.lazyProperty { SemanticVersion.release() }
-    var gradleVersion: SemanticVersion by gradleVersionProperty
-    private var config: ProjectConfig? = null
-
-    private var gradleVersionRange: VersionRange? = null
-
-    override val title: String = "Gradle"
-
-    override fun Panel.buildComponent(creator: MinecraftProjectCreator, context: WizardContext) {
-        row("Gradle version:") {
-            comboBox(model)
-                .bindItem(gradleVersionProperty)
-                .enabled(false) // TODO load compatible Gradle versions list
+class JdkProjectSetupFinalizer(
+    parent: NewProjectWizardStep
+) : AbstractNewProjectWizardStep(parent), ProjectSetupFinalizer {
+    private val sdkProperty: GraphProperty<Sdk?> = propertyGraph.property(null)
+    private var sdk by sdkProperty
+    private var sdkComboBox: JdkComboBox? = null
+
+    private var theMinVersion = JavaSdkVersion.JDK_17
+    var minVersion
+        get() = theMinVersion
+        set(version) {
+            if (version != theMinVersion) {
+                theMinVersion = version
+                sdkComboBox?.reloadModel()
+            }
         }
-    }
 
-    override fun isApplicable(creator: MinecraftProjectCreator, context: WizardContext): Boolean {
-        val buildSystem = creator.buildSystem
-        return buildSystem is GradleBuildSystem
+    init {
+        storeToData()
     }
 
-    override fun validateConfigs(creator: MinecraftProjectCreator, context: WizardContext): Boolean {
-        config = creator.config
-
-        if (creator.buildSystem !is GradleBuildSystem) {
-            return true
+    override fun setupUI(builder: Panel) {
+        with(builder) {
+            row("JDK Version:") {
```

Comment:
Maybe change `JDK Version :` to `JDK :` to fit with other types of project wizard.

![image](https://user-images.githubusercontent.com/40738104/215262989-43cf9753-8bc1-4a1b-bcb6-e4c44231a3f5.png)
![image](https://user-images.githubusercontent.com/40738104/215262995-1e94a30b-22e3-4505-825e-23d72461753e.png)